### DISCUSSION TITLE
Pivot Information of an Eloquent Collection is lost after a change in another property

### DISCUSSION THREAD

**Discussion by hmreumann** [livewire/livewire#3487] [7. Bugs (archived)]
Hallo, i think this is a strange behaviour.

On the mount() method I get the sharedUsers() of a Many to Many relationship.

    public function mount()
    {
        $this->sharedUsers = $this->exam->sharedUsers()->get();
    }

Then, after I make a change on another input field of a binded property "$searchUser", that has none relation with the previous.
When I make a change on that property the error is this:
" ErrorException
Trying to get property 'permissions' of non-object ..."

I checked with dd($this->sharedUsers->pivot->permissions) on the updatedSearchUser() and all the pivot was lost.

What I did was running again the mount command at the beginning of the updatedSearchUser () function. But it seems to be not efficient.

Thanks anyway. Hernán.

**Comment by ryatkins** [comment]
You can try a few things:

1) eager load your relationship in a separate method that is called throughout the lifecycle  (perhaps on mount, updating or instead on render)

or

2) use a computed property to eager load the relationship.  You won’t be able to bind to the property but it should preserve your relationship  https://laravel-livewire.com/docs/computed-properties/

**Reply by hmreumann** [reply]
Hallo, thanks for your reply. I'll try your suggestions.

**Comment by joshhanley** [comment]
@hmreumann pivot data is only accessible when you access it through a relation on another object.

Because you are assigning shared users to a local property, what is happening is when Livewire does a network request, it only keeps track of the class and ID's of the models in the collection, so when it rehydrates, it is asking the database for those exact records, hence why the pivot data isn't there.

If you need the pivot data, the best option would be to work from your `$this->exam` property to access shared users and the pivot data.

Hope this helps!

**Reply by hmreumann** [reply]
Yes, probably that Is a good solution. Thank you very much for your support.

**Reply by joshhanley** [reply]
@hmreumann no worries! 🙂
